### PR TITLE
Make it build with ghc-9.8

### PR DESCRIPTION
**Pull request by erikd**
# Description

Make it build with ghc-9.8

# Checklist

- Branch
    - [x] Updated changelog files.
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -53,7 +53,10 @@ package network-mux
 package ouroboros-network
   flags: +asserts +cddl
 
-if impl(ghc >= 9.6)
```
Comment (erikd) [inline cabal.project:56]: The old `if impl(ghc >= 9.6)` was no longer needed.

**Comment by erikd** [comment]
Does this PR really need an entry in all the CHNAGELOG files?

**Comment by erikd** [comment]
The **Haskell CI / build (9.8, windows-latest) (pull_request)** build is failing due the partial functions like `head` and `tail` now having a warning in `ghc-9.8` and  `-Werror` turning that into an error. 

The Nix job failure is due to a heap exhaustion error which is suspicious.

**Review by coot — APPROVED** [review]
LGTM

**Comment by coot** [comment]
> The Nix job failure is due to a heap exhaustion error which is suspicious.

We added heap limits quite recently and we're improving memory footprint of `io-sim` & our test suite.  Sometimes on larger test cases the memory consumption is much higher that average.

**Review by coot — APPROVED** [review]
LGTM

**Pull request merged**